internal/bminventory/inventory.go

```diff
@@ -4940,6 +4940,15 @@ func (b *bareMetalInventory) UpdateInfraEnvInternal(ctx context.Context, params
 		params.InfraEnvUpdateParams.PullSecret = pullSecretBackup
 	}
 
+	// The OpenShift installer validation code for the additional trust bundle
+	// is buggy and doesn't react well to additional newlines at the end of the
+	// certs. We need to strip them out to not bother assisted users with this
+	// quirk.
+	if params.InfraEnvUpdateParams.AdditionalTrustBundle != nil {
+		additionalTrustBundle := strings.TrimSpace(*params.InfraEnvUpdateParams.AdditionalTrustBundle)
+		params.InfraEnvUpdateParams.AdditionalTrustBundle = &additionalTrustBundle
+	}
+
```

Comment:
Why is this code copy-pasted here?

Reply:
The following line is b.validateAndUpdateInfraEnvParams() which contains the exact same code.  What am I missing here?

Reply:
You're not missing anything. I missed that.

@paul-maidment can you open a followup to further deduplicate this logic?

Reply:
Where exactly is the duplication, I'm not seeing it?

Reply:
Addressed in https://github.com/openshift/assisted-service/pull/5372